src/Zip.Shared/ZipFile.Check.cs

```diff
@@ -128,57 +128,59 @@ public static bool CheckZip(string zipFileName)
 
                 zip2 = ZipFile.Read(zipFileName);
 
-                foreach (var e1 in zip1)
+                foreach (var e2 in zip2)
                 {
-                    foreach (var e2 in zip2)
+                    try
                     {
-                        if (e1.FileName == e2.FileName)
+                        var e1 = zip1._entries[e2.FileName];
```

Comment:
Can we use TryGetValue here? That should be a lot better performance-wise (and in my opinion is more clear as well) than exception-based logic.